util/timer/src/timer.rs

```diff
@@ -42,7 +43,12 @@ pub struct TimerLoop {
 }
 
 impl TimerLoop {
-    pub fn new(worker_size: usize) -> TimerLoop {
+    pub fn new() -> TimerLoop {
+        let cpus = num_cpus::get();
```

Comment:
I don' think you need to create as many threads as the number of CPUs.
The TimerLoop is not the only thread CodeChain create.